File: app/src/main/java/br/com/nicetuber/feature/channel_details/ChannelDetailsPresenter.java
package br.com.nicetuber.feature.channel_details;

import javax.inject.Inject;

import br.com.nicetuber.base.BasePresenter;
import br.com.nicetuber.model.ChannelStatistics;

/**
 * Created by thiagozg on 15/10/2017.
 */

public class ChannelDetailsPresenter extends BasePresenter<IChannelDetails.View, ChannelDetailsInteractor>
    implements IChannelDetails.Callback {

    @Inject
    public ChannelDetailsPresenter() {
    }

    public void getChannelStatistics(String channelId) {
        interactor.getChannelStatistics(this, channelId);
    }

    @Override
    public void getChannelStatisticsSuccess(ChannelStatistics channelStatistics) {
        view.showChannelStatistics(channelStatistics);
    }

    @Override
    public void getChannelStatisticsError(String message) {
        view.showErrorMessage(message);
    }

}

File: app/src/main/java/br/com/nicetuber/feature/channel_details/ChannelDetailsPresenter.kt
package br.com.nicetuber.feature.channel_details

import br.com.nicetuber.base.BasePresenter
import br.com.nicetuber.model.ChannelStatistics
import javax.inject.Inject

/**
 * Created by thiagozg on 15/10/2017.
 */

class ChannelDetailsPresenter @Inject
constructor() : BasePresenter<IChannelDetails.View, ChannelDetailsInteractor>(), IChannelDetails.Callback {

    fun getChannelStatistics(channelId: String) {
        interactor.getChannelStatistics(this, channelId)
    }

    override fun getChannelStatisticsSuccess(channelStatistics: ChannelStatistics) {
        view?.showChannelStatistics(channelStatistics)
    }

    override fun getChannelStatisticsError(message: String) {
        view?.showErrorMessage(message)
    }

}
